php/commands/widget.php

```diff
@@ -305,6 +305,81 @@ public function delete( $args, $assoc_args ) {
 	}
 
 	/**
+	 * Reset sidebar.
+	 *
+	 * Removes all widgets from the sidebar and place those to Inactive Widgets.
+	 *
+	 * [<sidebar-id>...]
+	 * : One or more sidebars to reset.
+	 *
+	 * [--all]
+	 * : If set, all sidebars will be reset.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Reset a sidebar
+	 *     $ wp widget reset sidebar-1
+	 *     Success: Sidebar 'sidebar-1' reset.
+	 *
+	 *     # Reset multiple sidebars
+	 *     $ wp widget reset sidebar-1 sidebar-2
+	 *     Success: Sidebar 'sidebar-1' reset.
+	 *     Success: Sidebar 'sidebar-2' reset.
+	 *
+	 *     # Reset all sidebars
+	 *     $ wp widget reset --all
+	 *     Success: Sidebar 'sidebar-1' reset.
+	 *     Success: Sidebar 'sidebar-2' reset.
+	 *     Success: Sidebar 'sidebar-3' reset.
+	 */
+	public function reset( $args, $assoc_args ) {
+
+		global $wp_registered_sidebars;
+
+		$all = \WP_CLI\Utils\get_flag_value( $assoc_args, 'all', false );
+
+		// Bail if no arguments and no all flag.
+		if ( ! $all && empty( $args ) ) {
+			WP_CLI::error( 'Please specify one or more sidebars, or use --all.' );
+		}
+
+		// Fetch all sidebars if all flag is set.
+		if ( $all ) {
+			$args = array_keys( $wp_registered_sidebars );
+		}
+
+		// Sidebar ID wp_inactive_widgets is reserved by WP core for inactive widgets.
+		if ( isset( $args['wp_inactive_widgets'] ) ) {
+			unset( $args['wp_inactive_widgets'] );
+		}
+
+		// Check if no registered sidebar.
+		if ( empty( $args ) ) {
```

Comment:
How would you get this error, if we're already checking for `empty( $args )` above?